sql/wsrep_high_priority_service.cc

```diff
@@ -467,6 +490,13 @@ int Wsrep_applier_service::apply_write_set(const wsrep::ws_meta& ws_meta,
   DBUG_ENTER("Wsrep_applier_service::apply_write_set");
   THD* thd= m_thd;
 
+  // The following line is a workaround to avoid a segfault
+  // in rpl_sql_thread_info while applying log events for XA.
+  // XA TODO figure out what causes the segfault and figure
+  // out an fix.
+  thd->system_thread_info.rpl_sql_info=
```

Comment:
We unconditionally allocate memory here is this NULL even when log-bin is enabled ? When/where it will be this memory released ?